schema/relative-path.json

```diff
@@ -0,0 +1,9 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "File path",
+    "description": "A relative cross-platform file path",
+    "version": "0.1.1",
+
+    "type": "string",
+    "pattern": "^([^<>:\"/\\\\|?*\\x00-\\x1f]+/)*[^<>:\"/\\\\|?*\\x00-\\x1f]+$"
```

Comment:
My reasoning behind the pattern was more trying to match the Windows technical limitation rather than being prescriptive about what we think a good filename should be.

`:cntrl:` is probably a good choice as it only includes one extra character (ESC) compared to `\x00-\x1f`, and I'm happy to exclude that character too. On the other hand, if people want to use weird printable characters (eg. `@~#`) I don't care as long as Windows supports it.

Reply:
So apparently Python's `re` module doesn't even support the named character classes such as `:alnum:`, and neither does the jsonschema spec (which defines regexes as whatever the ECMAScript standard says).

ECMAScript does allow the "unicode property" syntax, ie. `\p{Cc}` for control characters, but the Python module does not, so I think we're stuck with explicit byte values for now.